src/main/js/parser.js

```diff
@@ -0,0 +1,98 @@
+import sax from "sax";
+
+/**
+ * @typedef {sax.Tag | sax.QualifiedTag} Node
+ */
+
+/**
+ * @typedef {Object} Parser
+ * @property {(xml: string) => Parser} write
+ * @property {() => Parser} close
+ * @property {(node: Node) => void} onopentag
+ * @property {(text: string) => void} ontext
+ * @property {() => void} onclosetag
+ */
+
+export class XMLParser {
```

Comment:
To avoid a massive refactor of `fromXML()`, the parser interface matches sax interface, and the `XMLParser` simply crawls the result of `DOMParser` and mimics the sax behavior.